fix for ISO warning

Fixing up some esoteric messages from the compiler and I came across this one.
```
vtr-vpr/libs/libvtrutil/src/vtr_ndmatrix.h:79:24: warning: template-id not allowed for constructor in C++20 [-Wtemplate-id-cdtor]
   79 |     NdMatrixProxy<T, 1>(const size_t* dim_sizes, const size_t* dim_stride, T* start)
      |                        ^
```

Here is the simple fix from my tree:
```
diff --git a/src/vtr-vpr/libs/libvtrutil/src/vtr_ndmatrix.h b/src/vtr-vpr/libs/libvtrutil/src/vtr_ndmatrix.h
index c3a4692..b34a9c8 100644
--- a/src/vtr-vpr/libs/libvtrutil/src/vtr_ndmatrix.h
+++ b/src/vtr-vpr/libs/libvtrutil/src/vtr_ndmatrix.h
@@ -34,7 +34,7 @@ class NdMatrixProxy {
      *    @param dim_stride: The stride of this dimension (i.e. how many element in memory between indicies of this dimension)
      *    @param  start: Pointer to the start of the sub-matrix this proxy represents
      */
-    NdMatrixProxy<T, N>(const size_t* dim_sizes, const size_t* dim_strides, T* start)
+    NdMatrixProxy(const size_t* dim_sizes, const size_t* dim_strides, T* start)
         : dim_sizes_(dim_sizes)
         , dim_strides_(dim_strides)
         , start_(start) {}
@@ -76,7 +76,7 @@ class NdMatrixProxy<T, 1> {
      *    @param dim_stride: The stride of this dimension (i.e. how many element in memory between indicies of this dimension)
      *    @param  start: Pointer to the start of the sub-matrix this proxy represents
      */
-    NdMatrixProxy<T, 1>(const size_t* dim_sizes, const size_t* dim_stride, T* start)
+    NdMatrixProxy(const size_t* dim_sizes, const size_t* dim_stride, T* start)
         : dim_sizes_(dim_sizes)
         , dim_strides_(dim_stride)
         , start_(start) {}

```

Sorry that this is an issue and not a PR; I'm juggling a lot of things and wanted to just communicate this small thing.

--- a/libs/libvtrutil/src/vtr_ndmatrix.h
+++ b/libs/libvtrutil/src/vtr_ndmatrix.h
@@ -34,12 +34,12 @@ class NdMatrixProxy {
      *    @param dim_stride: The stride of this dimension (i.e. how many element in memory between indicies of this dimension)
      *    @param  start: Pointer to the start of the sub-matrix this proxy represents
      */
-    NdMatrixProxy<T, N>(const size_t* dim_sizes, const size_t* dim_strides, T* start)
+    NdMatrixProxy(const size_t* dim_sizes, const size_t* dim_strides, T* start)
         : dim_sizes_(dim_sizes)
         , dim_strides_(dim_strides)
         , start_(start) {}
 
-    NdMatrixProxy<T, N>& operator=(const NdMatrixProxy<T, N>& other) = delete;
+    NdMatrixProxy& operator=(const NdMatrixProxy& other) = delete;
 
     ///@brief const [] operator
     const NdMatrixProxy<T, N - 1> operator[](size_t index) const {
@@ -76,12 +76,12 @@ class NdMatrixProxy<T, 1> {
      *    @param dim_stride: The stride of this dimension (i.e. how many element in memory between indicies of this dimension)
      *    @param  start: Pointer to the start of the sub-matrix this proxy represents
      */
-    NdMatrixProxy<T, 1>(const size_t* dim_sizes, const size_t* dim_stride, T* start)
+    NdMatrixProxy(const size_t* dim_sizes, const size_t* dim_stride, T* start)
         : dim_sizes_(dim_sizes)
         , dim_strides_(dim_stride)
         , start_(start) {}
 
-    NdMatrixProxy<T, 1>& operator=(const NdMatrixProxy<T, 1>& other) = delete;
+    NdMatrixProxy& operator=(const NdMatrixProxy& other) = delete;
 
     ///@brief const [] operator
     const T& operator[](size_t index) const {
@@ -407,3 +407,4 @@ using Matrix = NdMatrix<T, 2>;
 
 } // namespace vtr
 #endif
+

